Makefile

```diff
@@ -34,35 +47,180 @@ VIM_PROG    := $(shell which vim)
 GIT_PROG    := $(shell which git)
 endif
 
-all: test generate
+DATA_HOME := $(XDG_DATA_HOME)
+DATA_HOME ?= '~/.local/share'
 
-test: test-vim test-nvim
+PHONY :=
 
-generate:
-ifdef PYTHON_PROG
-	$(PYTHON_PROG) scripts/gen_code_completion.py
+# ----------------------------------------------------------------------------
+#
+# General
+#
+# These targets are what you'll probably need the most.
+#
+# ----------------------------------------------------------------------------
+# If make is invoked without any target provided, this will be our default
+# target.
+PHONY += help
+help:
+	@echo	'Cleaning:'
+	@echo	'	clean		- Remove all dependencies located in `.bundle/`.'
+	@echo	''
+	@echo	'Development:'
+	@echo	'	test		- Run test suite for Neovim and Vim.'
+	@echo	'				  This target automatically skips Neovim and/or Vim'
+	@echo	'				  if the respective program is not found.'
+	@echo	'	test-vim	- Run test suite only for Vim.'
```

Comment:
I know GitHub renders tabs weirdly here, but this does display properly in the terminal.